File: src/main/java/chinchulin/varano/Models/AcademicRecord.java
package chinchulin.varano.Models;

import chinchulin.varano.Utils.AcademicState;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "academic_records")
public class AcademicRecord {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id_academic_record;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "id_student")
    private Student student;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "id_course")
    private Course course;

    @Column(name = "study_year")
    private Short studyYear;

    @Enumerated(EnumType.STRING)
    private AcademicState state = AcademicState.REGULAR;

    @Column(name = "average", nullable = false)
    private Double average = 0d;

    @Column(name = "unique_record_code", nullable = false, unique = true)
    private String uniqueCode;

    @Column(name = "referred_comment")
    private String comment = null;
}

File: src/main/kotlin/com/project/varano/models/AcademicRecord.kt
package com.project.varano.models

import com.project.varano.enums.AcademicState
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.EnumType
import jakarta.persistence.Enumerated
import jakarta.persistence.FetchType
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne
import jakarta.persistence.Table
import lombok.AllArgsConstructor
import lombok.Builder
import lombok.Data
import lombok.NoArgsConstructor

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "academic_records")
data class AcademicRecord (
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_academic_record")
    private val id: Long,

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "id_student")
    private val student: Student,

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "id_course")
    private val course: Course,

    @Column(name = "study_year")
    private val studyYear: Short,

    @Enumerated(EnumType.STRING)
    private val state: AcademicState = AcademicState.REGULAR,

    @Column(nullable = false)
    private val average: Float = 0.0f,

    @Column(name = "unique_record_code", nullable = false, unique = true)
    private val uniqueCode: String,

    @Column(name = "referred_comment")
    private val comment: String? = null
)
